backend/ml/outputs.py

```diff
@@ -0,0 +1,113 @@
+import pandas as pd
+from django.db import transaction
+
+from core.models import User
+from tournesol.models import (
+    ContributorRating,
+    ContributorRatingCriteriaScore,
+    Entity,
+    EntityCriteriaScore,
+)
+
+
+def save_entity_scores(poll, entity_scores, single_criteria=None):
+    if isinstance(entity_scores, pd.DataFrame):
+        scores_iterator = entity_scores[
+            ["entity_id", "criteria", "score", "uncertainty"]
+        ].itertuples(index=False)
+    else:
+        scores_iterator = entity_scores
+
+    with transaction.atomic():
+        scores_to_delete = EntityCriteriaScore.objects.filter(poll=poll)
+        if single_criteria:
+            scores_to_delete = scores_to_delete.filter(criteria=single_criteria)
+        scores_to_delete.delete()
+
+        EntityCriteriaScore.objects.bulk_create(
+            EntityCriteriaScore(
+                poll=poll,
+                entity_id=entity_id,
+                criteria=criteria,
+                score=score,
+                uncertainty=uncertainty,
+            )
+            for entity_id, criteria, score, uncertainty in scores_iterator
+        )
+
+
+def save_tournesol_score_as_sum_of_criteria(poll):
+    entities = []
+    for entity in (
+        Entity.objects.filter(criteria_scores__poll=poll)
+        .distinct()
+        .prefetch_related("criteria_scores")
+    ):
+        entity.tournesol_score = 10 * sum(
+            criterion.score for criterion in entity.criteria_scores.all()
+        )
+        entities.append(entity)
+    Entity.objects.bulk_update(entities, ["tournesol_score"])
+
+
+def save_contributor_scores(
+    poll, contributor_scores, trusted_filter=None, single_criteria=None
+):
+    if isinstance(contributor_scores, pd.DataFrame):
+        scores_list = list(
+            contributor_scores[
+                ["user_id", "entity_id", "criteria", "score", "uncertainty"]
+            ].itertuples(index=False)
+        )
+    else:
+        scores_list = list(contributor_scores)
+
+    rating_ids = {
+        (contributor_id, video_id): rating_id
+        for rating_id, contributor_id, video_id in ContributorRating.objects.filter(
+            poll=poll
+        ).values_list("id", "user_id", "entity_id")
+    }
```

Comment:
```suggestion
    rating_ids = {
        (contributor_id, video_id): rating_id 
        for rating_id, contributor_id, video_id in ContributorRating.objects
            .filter(poll=poll).values_list("id", "user_id", "entity_id")
    }
```

Reply:
Saw with Adrien.

Thanks again for the feedback.

We're going to keep this as is for now, as this suggested change will break the consistency between this file and the rest of the module / project.

We will consider updating those line breaks after the deliberation about the module `black`.